packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import memoize from 'memize';
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Children, cloneElement, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../inspector-controls';
+import PanelColorSettings from '../panel-color-settings';
+import { useBlockEditContext } from '../block-edit';
+
+const createComponent = memoize( ( attribute, color ) => ( { children } ) =>
+	// Clone children, setting the style attribute from the color configuration,
+	// if not already set explicitly through props.
+	Children.map( children, ( child ) =>
+		cloneElement( child, {
+			style: { [ attribute ]: color, ...child.props.style },
+		} )
+	)
+);
+
+const createSetColor = memoize( ( setAttributes, name ) => ( newColor ) =>
+	setAttributes( { [ name ]: newColor } )
+);
+
+const InspectorControlsColorPanel = ( { title, colorSettings } ) => (
+	<InspectorControls>
+		<PanelColorSettings
```

Comment:
We should allow the hook users to add children that appear inside PanelColorSettings and these children them selfs may depend on the color values. E.g: it is very common to add contrast checking UI inside this panel.

Reply:
Good idea.

Reply:
Can't we back contract checking into the hook somehow?

Reply:
>back contract checking

Not sure what that means.

Reply:
Oh got it.

Reply:
Yeah we could, make it an option and turn it on by default.

Reply:
I'll add it to the "next steps".